pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -152,3 +167,130 @@ func (c configResource) computeAndReconcileConfigMap(xsite *configuration.XSite)
 	}
 	return err
 }
+
+// addInfinispanConfig installs a reconciler for a specific object.
+// Only events coming from operatorNs/infinispan-operator-config will be processed
+func addInfinispanConfig(mgr manager.Manager) error {
+	r := newInfinispanOperatorConfigReconciler(mgr)
+	// Create a new controller
+	c, err := controller.New("infinispan-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	operatorNS, err := getOperatorNamespace()
+	if err != nil && err != k8sutil.ErrRunLocal {
+		return err
+	}
+	operatorConfigPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetName() == "infinispan-operator-config" && e.MetaNew.GetNamespace() == operatorNS
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+	}
+	// Watch for changes to global operator config
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, operatorConfigPredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// newInfinispanOperatorConfigReconciler returns a new reconcile.Reconciler
+func newInfinispanOperatorConfigReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileInfinispanOperatorConfig{client: mgr.GetClient(), scheme: mgr.GetScheme(), Log: logf.Log.WithName("operator-config-controller"), Kube: kube.NewKubernetesFromController(mgr)}
+}
+
+var _ reconcile.Reconciler = &ReconcileInfinispanOperatorConfig{}
+
+// ReconcileInfinispan reconciles a Infinispan object
+type ReconcileInfinispanOperatorConfig struct {
+	client client.Client
+	scheme *runtime.Scheme
+	Log    logr.Logger
+	Kube   *kube.Kubernetes
+}
+
+// Reconcile reads the operator configuration in the `infinispan-operator-config` ConfigMap and
+// applies the configuration to the cluster.
+func (r *ReconcileInfinispanOperatorConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
As we have a reconcile method that is independent of the existing `Process()` method used by the `Reconciler` inteface, maybe it's best to have the operator configuration controller as a separate file? It seems like their is minimal shared code between the two atm and the separation would make things much easier to read.